test/jdk/java/nio/channels/FileChannel/MapTest.java

```diff
@@ -56,21 +57,42 @@ public class MapTest {
     public static void main(String[] args) throws Exception {
         blah = File.createTempFile("blah", null);
         blah.deleteOnExit();
+        long t0 = System.nanoTime();
         initTestFile(blah);
+        long t1 = System.nanoTime();
+        out.printf("Test file %s initialized in %d ns (%d ms) %n",
+                blah, t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+        t0 = t1;
         try {
-            out.println("Test file " + blah + " initialized");
             testZero();
-            out.println("Zero size: OK");
+            t1 = System.nanoTime();
+            out.printf("Zero size: done in %d ns (%d ms) %n",
+                    t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+            t0 = t1;
             testRead();
-            out.println("Read: OK");
+            t1 = System.nanoTime();
+            out.printf("Read: done in %d ns (%d ms) %n",
+                    t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+            t0 = t1;
             testWrite();
-            out.println("Write: OK");
+            t1 = System.nanoTime();
+            out.printf("Write: done in %d ns (%d ms) %n",
+                    t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+            t0 = t1;
             testHighOffset();
-            out.println("High offset: OK");
+            t1 = System.nanoTime();
+            out.printf("High offset: done in %d ns (%d ms) %n",
+                    t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+            t0 = t1;
             testForce();
-            out.println("Force: OK");
+            t1 = System.nanoTime();
+            out.printf("Force: done in %d ns (%d ms) %n",
+                    t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+            t0 = t1;
             testExceptions();
-            out.println("Exceptions: OK");
+            t1 = System.nanoTime();
+            out.printf("Exceptions: done in %d ns (%d ms) %n",
+                    t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
```

Comment:
Are you planning to leave all these timestamps in the output? I can't tell if they are left over from your testing or intentional.

Reply:
Yeah, I wanted to leave them here; they don't take up much space, and are very useful when trying to figure out why the tests are slow.